pcsx2/gui/Dialogs/ModalPopups.h

```diff
@@ -20,54 +20,344 @@
 #include "Panels/ConfigurationPanels.h"
 
 #include <wx/wizard.h>
+#include <wx/treectrl.h>
+
+#define GSDUMP_GIFREG \
+X(PRIM,0x00) \
+X(RGBAQ,0x01) \
+X(ST,0x02) \
+X(UV,0x03) \
+X(XYZF2,0x04) \
+X(XYZ2,0x05) \
+X(TEX0_1,0x06) \
+X(TEX0_2,0x07) \
+X(CLAMP_1,0x08) \
+X(CLAMP_2,0x09) \
+X(FOG,0x0a) \
+X(XYZF3,0x0c) \
+X(XYZ3,0x0d) \
+X(AD,0x0e) \
+X(NOP,0x0f) \
+X(TEX1_1,0x14) \
+X(TEX1_2,0x15) \
+X(TEX2_1,0x16) \
+X(TEX2_2,0x17) \
+X(XYOFFSET_1,0x18) \
+X(XYOFFSET_2,0x19) \
+X(PRMODECONT,0x1a) \
+X(PRMODE,0x1b) \
+X(TEXCLUT,0x1c) \
+X(SCANMSK,0x22) \
+X(MIPTBP1_1,0x34) \
+X(MIPTBP1_2,0x35) \
+X(MIPTBP2_1,0x36) \
+X(MIPTBP2_2,0x37) \
+X(TEXA,0x3b) \
+X(FOGCOL,0x3d) \
+X(TEXFLUSH,0x3f) \
+X(SCISSOR_1,0x40) \
+X(SCISSOR_2,0x41) \
+X(ALPHA_1,0x42) \
+X(ALPHA_2,0x43) \
+X(DIMX,0x44) \
+X(DTHE,0x45) \
+X(COLCLAMP,0x46) \
+X(TEST_1,0x47) \
+X(TEST_2,0x48) \
+X(PABE,0x49) \
+X(FBA_1,0x4a) \
+X(FBA_2,0x4b) \
+X(FRAME_1,0x4c) \
+X(FRAME_2,0x4d) \
+X(ZBUF_1,0x4e) \
+X(ZBUF_2,0x4f) \
+X(BITBLTBUF,0x50) \
+X(TRXPOS,0x51) \
+X(TRXREG,0x52) \
+X(TRXDIR,0x53) \
+X(HWREG,0x54) \
+X(SIGNAL,0x60) \
+X(FINISH,0x61) \
+X(LABEL,0x62)
+
+#define X(name, value) name = value,
```

Comment:
I think it'll be a bit more obvious what's going on if you do
```c
#define GSDUMP_GIFREG(X) \
	X(PRIM, 0x00)
```
etc

Then, don't define the `X` until you're about to instantiate, at which point you do

```c
enum GSDUMP_GIFREG_NAME : GSDUMP_GIFREG_TYPE
{
#define ENTRY(name, value) name = value,
 	GSDUMP_GIFREG(ENTRY)
#undef ENTRY
};
```

Also I would like it if you indent the entries and align the numbers like
```c
#define GSDUMP_GIFREG(X) \
	X(PRIM,       0x00) \
	X(RGBAQ,      0x01) \
	X(ST,         0x02) \
	X(UV,         0x03) \
	X(XYZF2,      0x04) \
	X(XYZ2,       0x05) \
	X(TEX0_1,     0x06) \
	X(TEX0_2,     0x07) \
	X(CLAMP_1,    0x08) \
	X(CLAMP_2,    0x09) \
	X(FOG,        0x0a) \
	X(XYZF3,      0x0c) \
	X(XYZ3,       0x0d) \
	X(AD,         0x0e) \
	X(NOP,        0x0f) \
	X(TEX1_1,     0x14) \
	X(TEX1_2,     0x15) \
	X(TEX2_1,     0x16) \
	X(TEX2_2,     0x17) \
	X(XYOFFSET_1, 0x18) \
	X(XYOFFSET_2, 0x19) \
	X(PRMODECONT, 0x1a) \
	X(PRMODE,     0x1b) \
	X(TEXCLUT,    0x1c) \
	X(SCANMSK,    0x22) \
	X(MIPTBP1_1,  0x34) \
	X(MIPTBP1_2,  0x35) \
	X(MIPTBP2_1,  0x36) \
	X(MIPTBP2_2,  0x37) \
	X(TEXA,       0x3b) \
	X(FOGCOL,     0x3d) \
	X(TEXFLUSH,   0x3f) \
	X(SCISSOR_1,  0x40) \
	X(SCISSOR_2,  0x41) \
	X(ALPHA_1,    0x42) \
	X(ALPHA_2,    0x43) \
	X(DIMX,       0x44) \
	X(DTHE,       0x45) \
	X(COLCLAMP,   0x46) \
	X(TEST_1,     0x47) \
	X(TEST_2,     0x48) \
	X(PABE,       0x49) \
	X(FBA_1,      0x4a) \
	X(FBA_2,      0x4b) \
	X(FRAME_1,    0x4c) \
	X(FRAME_2,    0x4d) \
	X(ZBUF_1,     0x4e) \
	X(ZBUF_2,     0x4f) \
	X(BITBLTBUF,  0x50) \
	X(TRXPOS,     0x51) \
	X(TRXREG,     0x52) \
	X(TRXDIR,     0x53) \
	X(HWREG,      0x54) \
	X(SIGNAL,     0x60) \
	X(FINISH,     0x61) \
	X(LABEL,      0x62)
```

Reply:
good idea! For the def/undef X and using it as an argument I feel like it's a matter of personal preference but I don't have anything against that.

Thanks for doing the indentation, definitely more readable this way :)